src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -134,7 +134,7 @@ protected static Object invokeMethod(Method thisMethod, Object instance, Object[
       }
     }
     throw new TestNGException("Data Provider " + dataProvider + " must return"
-          + " either Object[][] or Iterator<Object>[], not " + dataProvider.getReturnType());
+          + " either Object[][] or Object[] or Iterator<Object>[], not " + dataProvider.getReturnType());

```

Comment:
In fact, since 6.10 and only `Object[]` is supported and a discussion is still opened for `Object` on the issue.
For reference: https://github.com/cbeust/testng/issues/1139